test/workflow_test.go

```diff
@@ -573,6 +575,48 @@ func (w *Workflows) WorkflowWithLocalActivityCtxPropagation(ctx workflow.Context
 	return result, nil
 }
 
+func (w *Workflows) NonDeterminismSimulatorWorkflow(ctx workflow.Context) ([]string, error) {
+	logger := workflow.GetLogger(ctx).Sugar()
+	ctx = workflow.WithActivityOptions(ctx, w.defaultActivityOptions())
+
+	workflow.SetQueryHandler(ctx, "custom_query", func() (int, error) {
+		return w.nonDeterminismSimulatorWorkflowCallCount, nil
+	})
+
+	res := []string{}
```

Comment:
```suggestion
	var res []string
```
for style points / zero allocation if unused / healthier habits.